File: 7 kyu/Doubleton number/solution.cs
using System.Linq;

public static class Kata
{
  public static int Doubleton(int num)
  {
    while ((++num).ToString().Distinct().Count() != 2);
    return num;
  }
}

###################
using System;
using System.Linq;

public static class Kata
{
    public static int Doubleton(int num)
    {
            num += 1;
            char[] numToArray = num.ToString().Distinct().ToArray();

            while (numToArray.Length != 2)
            {
                num += 1;
                numToArray = num.ToString().Distinct().ToArray();
            }

            return num;
    }
}

##################
using System.Linq;

public static class Kata
{
    public static int Doubleton(int num)
    {
        while ($"{++num}".Distinct().Count() != 2); return num;
    }
}

###################
using System;
public static class Kata
{
    public static int Doubleton(int num)
    {
      if (num < 10) return 10;
     for (int i = num + 1;;i++) 
     {
      if (IsDoubleton(i))
      {
        return i;
      }
     }
    }
    public static bool IsDoubleton(int num)
    {
        string diżit = Convert.ToString(num);
        char dubA = diżit[0], dubB = '\0';
        for (int i = 1; i < diżit.Length; i++)
        {
          char ch = diżit[i];
          if (ch != dubA)
          {
            dubB = ch;
            break;
          }
          
        }
      if (dubB == '\0')
      {
        return false;
      }
        foreach (char ch in diżit)
        {
         if (ch != dubA && ch != dubB) 
           return false;
        }
      return true;
    }
}

#########################
using System.Linq;
public static class Kata
{
    public static int Doubleton(int n) 
    {
      while ((++n).ToString().Distinct().Count() != 2);
      return n;
    } 
}

###################
using System.Linq;
public static class Kata
{
    public static int Doubleton(int num)
    {
      int nextNum = num+1;
        while(!isDoubleton(nextNum)){
          ++nextNum;
        }
      return nextNum;
    }
  private static bool isDoubleton(int num){
    return num.ToString().Distinct().Count()==2;
  }
}

##################
using System;
using System.Linq;

public static class Kata
{
    public static int Doubleton(int num)
    {
        for(int i =num+1; i< 5000000; i++){
          int check =i.ToString().Distinct().ToArray().Length;
          if(check ==2){return i;}
        }
      return -1;
    }
}

###################
using System.Linq;

public static class Kata
{
    public static int Doubleton(int num)
    {
        for (int i = num + 1; i < 1000000; i++)
            {
                if ((i.ToString().GroupBy(c => c).Count()) == 2)
                    return i;
            }
            return 10000000;
    }
}

######################
using System.Linq;

public static class Kata
{
    public static int Doubleton(int num)
    {
        int current = num+1;

        while(IsDoubleton(current) == false)            
            current++;            

        return current;
    }


    public static bool IsDoubleton(int num) =>        
        $"{num}"
            .Distinct()
            .Count() == 2;
}
##########################
using System.Linq;
class Kata
{
  public static int Doubleton(int n)
  {
      while ((++n + "").Distinct().Count() != 2){ }
      return n;
  }
}

####################
using System.Linq;

public static class Kata {
    public static int Doubleton(int num) {
      int nextInt = num + 1;
      while (!IsDoubleton(nextInt))
        nextInt += 1;
      return nextInt;
    }
    
    private static bool IsDoubleton(int num) {
      return num.ToString().Distinct().Count() == 2;
    }
}

File: 7 kyu/Doubleton number/solution.py
def doubleton(num):
    num += 1
    while True:
        num_set = set(str(num))
        if len(num_set) != 2:
            num += 1
        else:
            return num
print(doubleton(123))
print(doubleton(121))

#################
def doubleton(num):
    n = num + 1
    while len(set(str(n))) != 2:
        n += 1
    return n
  
###########
from itertools import count

def doubleton(num):
    return next(n for n in count(num+1) if len(set(str(n))) == 2)
  
#############
def doubleton(num):
    is_doubleton = False
    
    while not is_doubleton:
        num += 1
        if len(set([int(i) for i in str(num)])) == 2:
            is_doubleton = True
            
    return num
        
################
def doubleton(num):
    return next(n for n in range(num+1,num*10) if len(set(str(n)))==2)
  
###############
def doubleton(num):
    l=''
    while len(l)!=2:
        num+=1
        l=set(map(int,str(num))) 
    return num
  
################
def help(l):
    s = set()
    for item in l:
        if item not in s:
            s.add(item)
    if len(s)!=2:
        return False
    else:
        return True
    
def doubleton(num):
    num +=1
    while not help(str(num)):
        num +=1
    return num

#################
def doubleton(num):
    num += 1
    while len({*str(num)}) != 2:
        num += 1
    return num
  
 ################
def doubleton(num):
    num = str(num)
    n = set()
    for elem in num:
        n.add(elem)
    if len(n) == 2:
        num = int(num)+1
        num = str(num)
        n = set()
        for elem in num:
            n.add(elem)
        while len(n) != 2:
            num = int(num)+1
            num = str(num)
            n = set()
            for elem in num:
                n.add(elem)
    else:
        while len(n) != 2:
            num = int(num)+1
            num = str(num)
            n = set()
            for elem in num:
                n.add(elem)
    return int(num)
